src/Common/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -279,7 +279,8 @@ public override Task SendAsync(ArraySegment<byte> buffer, WebSocketMessageType m
 
             try
             {
-               WebSocketValidate.ThrowIfInvalidState(_state, _disposed, s_validSendStates);
+                ThrowIfDisposed();
```

Comment:
I checked the access modifiers for functions like `SendAsync` in ManagedWebSocket, and found they are all `public`. That's why I have the concern first. 

I have checked the code, and can confirm that caller doesn't have direct access to the ManagedWebSocket. I have opened issue #23027 to keep track of this.